store-implementation/proxy-store/src/main/java/uk/gov/gchq/gaffer/proxystore/response/deserialiser/impl/DefaultResponseDeserialiser.java

```diff
@@ -36,8 +42,21 @@ public O deserialise(final String jsonString) throws SerialisationException {
             // The input is likely a plain java.lang.String object, so return as-is
             return (O) jsonString;
         } else {
-            // The input is likely a valid JSON value type, so process using the deserialiser
-            return JSONSerialiser.deserialise(encodeString(jsonString), typeReference);
+            // For some operations like 'Limit' we may not know the output type up front so it
+            // will be left as a generic '?', this means some objects will not be deserialised correctly.
+            try {
+                // Check if we can deserialise to common outputs like list of elements.
+                // This is a workaround until proxy store has been refactored
+                if (typeReference.getType().getTypeName().equals(Iterable.class.getName() + "<?>")) {
+                    return (O) JSONSerialiser.deserialise(encodeString(jsonString), new TypeReferenceImpl.IterableElement());
+                } else {
+                    return JSONSerialiser.deserialise(encodeString(jsonString), typeReference);
```

Comment:
This line is duplicated in the catch, I'm wondering if it could be used in a `finally` instead?

Reply:
Think that would be incorrect use of a finally block as you are saying you always want to run whatever is in the finally block whereas, in this case we only want to run it conditionally as a fallback.

Reply:
I was thinking of something like this:
```
O rtn = null;
try {
    if (case) {
        rtn = ...
    }
} catch (e) {
    LOGGER.error(...)
} finally {
    if (rtn == null) {
        rtn == fallback
    }
    return rtn;
}
```
Though this might not be a great idea just to save some duplication.

Reply:
Fair I see what you mean, though I'm not sure if its really worth changing especially as this will all be made obsolete under proposed redesign of proxy store